plugins/Goals/Visualizations/GoalsPages.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\Goals\Visualizations;
+
+use Piwik\API\Request;
+use Piwik\Common;
+use Piwik\DataTable\Filter\AddColumnsProcessedMetricsGoal;
+use Piwik\Piwik;
+use Piwik\Plugins\CoreVisualizations\Visualizations\HtmlTable;
+use Piwik\Site;
+
+require_once PIWIK_INCLUDE_PATH . '/core/Twig.php';
+
+/**
+ * DataTable Visualization that derives from HtmlTable and shows goals metrics for page conversions
+ */
+class GoalsPages extends Goals
+{
+    const ID = 'tableGoalsPages';
+
+    public function beforeRender()
+    {
+        $this->removeExcludedColumns();
+
+        $this->config->addTranslation('nb_hits', Piwik::translate('General_ColumnUniquePageviews'));
+        $this->config->metrics_documentation['nb_hits'] = Piwik::translate('General_ColumnUniquePageviewsDocumentation');
+
+        parent::beforeRender();
+    }
+
+    protected function setPropertiesForGoalsOverview($idSite)
+    {
+        $allGoals = $this->getGoals($idSite);
+
+        // set view properties
+        $this->config->columns_to_display = array('label', 'nb_hits');
+
+        foreach ($allGoals as $goal) {
+            $column        = "goal_{$goal['idgoal']}_nb_conversion_page_rate";
+            $this->config->columns_to_display[]  = $column;
+        }
+
+    }
+
+    protected function setPropertiesForGoals($idSite, $idGoals)
+    {
+
+        $allGoals = $this->getGoals($idSite);
+
+        if ('all' == $idGoals) {
+            $idGoals = array_keys($allGoals);
+        } else {
+            // only sort by a goal's conversions if not showing all goals (for FULL_REPORT)
+            $this->requestConfig->filter_sort_column = 'goal_' . reset($idGoals) . '_nb_conversions';
```

Comment:
I've committed a fix for that, for specific goals on the page url report it was attempting to sort on conversions rather than conversions_float. For the goal overview where multiple goal columns are shown in the same table it should now sort on page views.